src/visitable.cpp

```diff
@@ -797,10 +806,10 @@ int visitable<inventory>::amount_of( const std::string& what, bool pseudo, int l
 
     long res = 0;
     for( const item *it : iter->second ) {
-        res += it->amount_of( what, pseudo, limit );
+        res = accumulate_with_limit( res, it->amount_of( what, pseudo, limit ), limit );
```

Comment:
Until ```+``` and ```-``` stop meaning the same thing, such as if additions are truncated if they reach some value.  Even if we add a numeric type that enforces a limit the same way this does, I wouldn't add operator overloads that don't act like the standard operations.